functionary/core/utils/workflow.py

```diff
@@ -91,11 +91,15 @@ def move_step(step: WorkflowStep, next: WorkflowStep | None = None) -> None:
         raise ValueError("Provided step must be a member of the same Workflow")
 
     with transaction.atomic():
-        if old_before_step := WorkflowStep.objects.filter(next=step).first():
+        if old_before_step := WorkflowStep.objects.filter(
+            workflow=step.workflow, next=step
+        ).first():
             old_before_step.next = step.next
             old_before_step.save()
 
-        if new_before_step := WorkflowStep.objects.filter(next=next).first():
+        if new_before_step := WorkflowStep.objects.filter(
```

Comment:
The changes to this file fix a bug I ran into once the UI was in place to let me move things around.  Specifically, if next was None (meaning move the step to the end of the workflow) that was a non-unique check, and would return steps from other workflows.  This fixes the filter so that we're looking for the matching step for this workflow only.